src/mots/cli.py

```diff
@@ -0,0 +1,240 @@
+import sys
+from pathlib import Path
+
+from mots import module
+from mots.config import FileConfig
+from mots.directory import Directory
+from mots.logging import init_logging
+from mots.config import DEFAULT_CONFIG_FILEPATH
+
+from mots.utils import get_list_input
+
+logger = logging.getLogger(__name__)
+
+
+class CLI:
+    """Wrap methods and call with the correct arguments."""
+
+    @staticmethod
+    def init(args):
+        """Call `file_config.init` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.init()
+
+    @staticmethod
+    def ls(args):
+        """Call `module.ls` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.ls(file_config.config["modules"])
+
+    @staticmethod
+    def show(args):
+        """Call `module.show` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.show(file_config.config["modules"], args.module)
+
+    @staticmethod
+    def validate(args):
+        """Call `module.validate` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.validate(file_config.config, args.repo_path)
+
+    @staticmethod
+    def clean(args):
+        """Call `module.clean` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        return module.clean(file_config)
+
+    @staticmethod
+    def query(args):
+        """Call `directory.query` with correct arguments."""
+        file_config = FileConfig(Path(args.path))
+        file_config.load()
+        directory = Directory(file_config)
+        directory.load()
+        result, rejected = directory.query(*args.paths)
+        for path in result:
+            modules = result[path]
+            module_names = ",".join([m.machine_name for m in modules])
+            owners = ",".join([",".join(m.owners) for m in modules])
+            sys.stdout.write(f"{path}:{module_names}:{owners}\n")
+
+    @staticmethod
+    def add(args):
+        """Prompt user to add a new module."""
+        params = {
+            "machine_name": input("Enter machine name of new module: "),
+            "name": input("Enter a human readable name: "),
+            "owners": get_list_input("Enter a comma separated list of owners"),
+            "peers": get_list_input("Enter a comma separated list of peers"),
+            "includes": get_list_input(
+                "Enter a comma separated list of paths to include"
+            ),
+            "excludes": get_list_input(
+                "Enter a comma separated list of paths to exclude"
+            ),
+        }
+        parent = input("Enter a machine name of the parent module (optional): ") or None
+        file_config = FileConfig(Path(args.path))
+        module.add(params, file_config, parent=parent, write=True)
```

Comment:
In regular use, are you preferring that users make changes to the yaml directly, or via the CLI?

Reply:
It's really up to whoever is changing the file to make their own decisions. Either edit it manually, push, and let Lando/Phabricator do their thing and show the automatically parsed changes, or use `mots` CLI. There is no preference on my end, since `mots clean` will run regardless before things land.